examples/cpp_api/memfs.cc

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file   memfs.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2018-2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * When run, this program will create a simple 2D dense array on memfs,
+ * write some data to it, and read a slice of the data back.
+ *
+ * Note: memfs requires a global Context object
+ */
+
+#include <iostream>
+#include <tiledb/tiledb>
+
+using namespace tiledb;
+
+// Name of array.
+std::string array_name_("mem://quickstart_dense_array");
+
+// Example-global Context object.
+Context ctx_;
+
+void create_array(std::string array_name) {
+  // Create an empty dense 4x4 array with "rows" and "cols" on memfs
+  Domain domain(ctx_);
+  domain.add_dimension(Dimension::create<int>(ctx_, "rows", {{1, 4}}, 4))
+      .add_dimension(Dimension::create<int>(ctx_, "cols", {{1, 4}}, 4));
+  ArraySchema schema(ctx_, TILEDB_DENSE);
+  schema.set_domain(domain)
+      .set_order({{TILEDB_ROW_MAJOR, TILEDB_ROW_MAJOR}})
+      .add_attribute(Attribute::create<int>(ctx_, "a"));
+  Array::create(array_name, schema);
+}
+
+void write_array(std::string array_name) {
+  // Write data to the array
+  std::vector<int> data = {
+      1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16};
+
+  // Try writing with non-global context
+  try {
+    Context ctx_non_global;
+    Array array(ctx_non_global, array_name, TILEDB_WRITE);
+  } catch (std::exception& e) {
+  }
```

Comment:
Can we print an error message to demonstrate in the output that this exception was indeed thrown ?

Reply:
The console will output the exception but I'll add a message for users